katsdpcal/reduction.py

```diff
@@ -857,6 +913,56 @@ def flush_pipeline(ts, parameters, solution_stores):
         The pipeline parameters
     solution_stores : dict of :class:`~.CalSolutionStore`-like
         Solution stores for the capture block, indexed by solution type
+    num_chunks : int, optional
+        Group the frequency channels into this many sections to obtain
+        pointing fits
+
+    Returns
+    -------
+    Beam solutions saved to telstate
     """
-    logger.info('~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~')
-    logger.info('Flushed pipeline')
+    # TODO: num_chunks should enter through paramters
+    num_chunks = 16
+    # Extract bandpass gains from solution stores
+    b_solutions = solution_stores['B_POINTING'].get_range(start_time=0, end_time = time.time()).values
+    # Extract some some commonly used constants from the TS and parameters
+    # Middle time for each dump
+    mid_times = b_solutions.get_range(start_time = 0, end_time= time.time()).times
+    target_str = ts.get_range('cbf_target', et = mid_times[-1])[0][0]
```

Comment:
The usual pipeline has:
```Python
        t0 = data['times'][scan_slice.start]
        t1 = data['times'][scan_slice.stop - 1]
        n_times = scan_slice.stop - scan_slice.start

        #  target string contains: 'target name, tags, RA, DEC'
        target_str = ts.get_range('cbf_target', et=t0)[0][0]
        target = katpoint.Target(target_str)
```
We therefore want the target just before the start of the scan (`t0` above), so a better approach would be:
```suggestion
    target_str = ts.get_range('cbf_target', et = mid_times[0])[0][0]
```
There really should be only a single target for the entire set of solutions but we currently don't doublecheck that.